content/docs/extensions/pg_prewarm.md

```diff
@@ -0,0 +1,241 @@
+
+```sql
+SELECT indexname FROM pg_indexes WHERE tablename = 'your_table_name';
+```
+Replace `your_table_name` with the name of the table whose indexes you're interested in. Once you have the index name, you can then use `pg_prewarm` as shown above.
+
+Additionally, if you prefer to use the index's OID, you can find it using the `pg_class` system catalog. Here's how to find an index's OID:
+
+```sql
+SELECT oid FROM pg_class WHERE relname = 'index_name';
+```
+
+Then, you can use the OID with `pg_prewarm` like so:
+
+```sql
+SELECT pg_prewarm(your_index_oid);
+```
+
+## Check the proportion of a table loaded into cache
+
+In this example, you create a table, check its data size, run `pg_prewarm`, and then check to see how much of the table's data was loaded into memory.
+
+1. First, create a table and populate it with some data:
+
+    ```sql
+    CREATE TABLE t_test AS
+    SELECT * FROM generate_series(1, 1000000) AS id;
+    ```
+
+2. Check the size of the table:
+
+    ```sql
+    SELECT pg_size_pretty(pg_relation_size('t_test')) AS table_size_pretty,
+        pg_relation_size('t_test') AS table_size_bytes;
+    ```
+
+    This command returns the size of the table in both MB and bytes.
+
+    ```sql
+     table_size_pretty | table_size_bytes 
+    -------------------+------------------
+    35 MB              |         36700160
+    ```
+
+3. Load the table data into the Postgres buffer cache using `pg_prewarm`:
+
+    ```sql
+    SELECT pg_prewarm('public.t_test') AS blocks_loaded;
+    ```
+
+    This will output the number of blocks that were loaded:
+
+    ```sql
+    blocks_loaded 
+    ---------------
+            4480
+    ```
+
+4. To understand the calculation that follows, check the block size of your Postgres instance:
+
+    ```sql
+    SHOW block_size;
+    ```
+
+    The default block size in Postgres is 8192 bytes (8KB). We'll use this value in the next step.
+
+    ```sql
+    block_size 
+    ------------
+    8192
+    ```
+
+5. Calculate the total size of the data loaded into the cache using the block size and the number of blocks loaded:
+
+    ```sql
+    -- Assuming 4480 blocks were loaded (replace with your actual number from pg_prewarm output)
+    SELECT 4480 * 8192 AS loaded_data_bytes;
+    ```
+
+    You can now compare this value with the size of your table.
+
+    ```sql
+     loaded_data_bytes 
+    -------------------
+            36700160
+    ```
+
+    The values for the size of the table and the size of the data loaded into the buffer cache as shown in the example above match exactly, which is an ideal scenario. However, there are cases where these values might not match, indicating that not all the data was loaded into the buffer cache; for example:
```

Comment:
I found this section a bit dense. I don't know if it can be simplified; it may be the nature of the content. But something to consider.

Reply:
TY. I shortened it.